R/with.R

```diff
@@ -11,13 +11,13 @@
 #' @seealso [linters_with_tags], [linters_with_defaults] for creating linter lists.
 #' @examples
 #' # custom list of undesirable functions:
-#' #    remove sapply (using NULL)
-#' #    add cat (with a accompanying message),
-#' #    add print (unnamed, i.e. with no accompanying message)
-#' #    add return (as taken from all_undesirable_functions)
+#' #    remove `sapply` (using `NULL`)
+#' #    add `cat` (with an accompanying message),
+#' #    add `print` (unnamed, i.e. with no accompanying message)
+#' #    add `source` (as taken from `all_undesirable_functions`)
 #' my_undesirable_functions <- modify_defaults(
 #'   defaults = default_undesirable_functions,
-#'   sapply = NULL, "cat" = "No cat allowed", "print", all_undesirable_functions[["return"]]
+#'   sapply = NULL, "cat" = "No cat allowed", "print", all_undesirable_functions[["source"]]
```

Comment:
Gets rid of a warning in the [docs](https://lintr.r-lib.org/dev/reference/modify_defaults.html):

> Trying to remove ‘return’, which is not in `defaults`.

Reply:
Oh that warning looks like a bug, if it appears as you commented?!
At least the intention seemed to be to select return in addition to some other stuff.

Reply:
> Oh that warning looks like a bug, if it appears as you commented?!

Yes: https://lintr.r-lib.org/dev/reference/modify_defaults.html#ref-examples

> At least the intention seemed to be to select return in addition to some other stuff.

I think the warning makes sense, no? `return()` is indeed not included in the default list: 

``` r
library(lintr)

all_undesirable_functions[["return"]]
#> NULL
```

<sup>Created on 2022-10-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

The code should instead look like the following?

``` r
library(lintr)

my_undesirable_functions <- modify_defaults(
  defaults = default_undesirable_functions,
  sapply = NULL, "cat" = "No cat allowed", "print", "return"
)

names(my_undesirable_functions)
#>  [1] ".libPaths"     "attach"        "browser"       "cat"          
#>  [5] "debug"         "debugcall"     "debugonce"     "detach"       
#>  [9] "library"       "mapply"        "options"       "par"          
#> [13] "print"         "require"       "return"        "setwd"        
#> [17] "sink"          "source"        "Sys.setenv"    "Sys.setlocale"
#> [21] "trace"         "undebug"       "untrace"
```

<sup>Created on 2022-10-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
... the code should use a function from `setdiff(names(all_undesirable_functions), names(default_undesirable_functions))`.

Reply:
Oh, I see. Should we create a new issue to track this?

Reply:
You changed it to `"source"` here already, so this PR would close the issue, right?

Reply:
Yeah, true that. I misunderstood your comment. Resolving this.